components/datatypes/consentpreferences.schema.json

```diff
@@ -0,0 +1,337 @@
+      "title": "Preference timestamp",
+      "description": "Timestamp of this specific opt out or preference.",
+      "type": "string",
+      "format": "date-time"
+    },
+    "consentOption": {
+      "type": "object",
+      "properties": {
+        "xdm:choice": {
+          "$ref": "#/definitions/consentValue"
+        },
+        "xdm:basisOfProcessing": {
+          "$ref": "#/definitions/basisOfProcessing"
+        },
+        "xdm:timestamp": {
+          "$ref": "#/definitions/timestamp"
+        }
+      }
+    },
+    "marketingOption": {
+      "type": "object",
+      "properties": {
+        "xdm:choice": {
+          "$ref": "#/definitions/consentValue"
+        },
+        "xdm:basisOfProcessing": {
+          "$ref": "#/definitions/basisOfProcessing"
+        },
+        "xdm:timestamp": {
+          "$ref": "#/definitions/timestamp"
+        },
+        "xdm:reason": {
+          "title": "Opt out reason",
+          "type": "string",
+          "description": "Reason for opting out"
+        }
+      }
+    },
+    "consent-preferences": {
+      "properties": {
+        "xdm:privacyPreferences": {
+          "title": "Privacy Preferences",
+          "description": "Encapsulates data privacy preferences or consents.",
+          "type": "object",
+          "properties": {
+            "xdm:dataCollection": {
+              "title": "Data Collection",
+              "description": "Data collection is permitted",
+              "$ref": "#/definitions/consentOption"
+            },
+            "xdm:sellingOrSharingOfData": {
+              "title": "Selling or Sharing of Data",
+              "description": "Sharing or Selling of Data with/to 2nd or 3rd parties is permitted, such as in compliance with CCPA",
+              "$ref": "#/definitions/consentOption"
+            },
+            "xdm:anonymousAnalysis": {
+              "title": "Anonymous Analysis",
+              "description": "Data can be used for analysis that does not require the use of any type of ID (whether it can be tied to an individual or not)",
+              "$ref": "#/definitions/consentOption"
+            },
+            "xdm:pseudonymousAnalysis": {
+              "title": "Pseudonymous Analysis",
+              "description": "Data can be used for analysis that may include an ID that may or may not be tied to an actual person",
+              "$ref": "#/definitions/consentOption"
+            },
+            "xdm:deviceLinking": {
+              "title": "Device Linking",
+              "description": "Attempts to associate multiple devices used/owned by an individual/household/organization are permitted",
+              "$ref": "#/definitions/consentOption"
+            }
+          }
+        },
+        "xdm:personalizationPreferences": {
+          "title": "Personalization Preferences",
+          "description": "User's Personalization Preferences",
+          "type": "object",
+          "properties": {
+            "xdm:general": {
```

Comment:
this is really too "general" a name...

Reply:
Previously I called this "default" but that was deemed even worse. I could call it "generalPersonalization", but that seemed redundant since it is inside the personalization preferences section. Do you have another suggestion?

Reply:
OK - I'll live with it